### PR TITLE
[CI, testing] add SVM C and nu parameter check prevents seg fault sklearn < 1.2

### PR DESCRIPTION
**Pull request by icfaust**
### Description 
This adds the necessary checks to prevent seg faults caused by C and Nu out of bounds which are handled on libsvm/oneDAL side.  In sklearn >=1.2 there are checks on parameter constraints which prevent this check from occuring on sklearn side.

Upon request of @Vika-F to convert this to a true PR.

Fixes # - _issue number(s) if exists_

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

File: sklearnex/svm/nusvc.py [hunk 1]
```diff
@@ -83,6 +83,8 @@ def __init__(
     def fit(self, X, y, sample_weight=None):
         if sklearn_check_version("1.2"):
             self._validate_params()
+        elif self.nu <= 0 or self.nu > 1:
+            raise ValueError("Margin error upper bound outside range: nu <= 0 or nu > 1")
```
Comment (samir-nasibli) [inline sklearnex/svm/nusvc.py:87]: Doesn't it make sense move this part into dispatching check for the support.
I would prefer stock scikit-learn to handle error raising

Comment (icfaust) [reply sklearnex/svm/nusvc.py:87]: THIS PR IS JUST TO VERIFY AGAINST CI TO SEE IF WINDOWS SKLEARN 1.0, 1.1 SEG FAULTS CAN BE STOPPED ON PYTHON SIDE

Comment (samir-nasibli) [reply sklearnex/svm/nusvc.py:87]: Thank you for copying the PR description. I read it before watching it. I think it makes sense to respond to comments a little more politely. This repo is public, it makes sense to leave a comment in a more appropriate manner.

File: sklearnex/svm/nusvc.py [hunk 2]
```diff
@@ -83,6 +83,8 @@ def __init__(
     def fit(self, X, y, sample_weight=None):
         if sklearn_check_version("1.2"):
             self._validate_params()
+        elif self.nu <= 0 or self.nu > 1:
+            raise ValueError("nu <= 0 or nu > 1")
```
Comment (Vika-F) [inline sklearnex/svm/nusvc.py:87]: Please add a comment that those checks are added to fix the failing test. I.e. need a description why only this parameter is checked here. And others like accuracy, number of iterations, etc. - are not checked.

**Review by Vika-F — COMMENTED** [review]
Please add a some clarifications in the code and I'll approve.

**Review by Vika-F — APPROVED** [review]
LGTM

File: sklearnex/svm/svc.py [hunk 3]
```diff
@@ -85,6 +85,17 @@ def __init__(
...
+            raise ValueError("C <= 0")
+            # else if added to correct issues with
+            # sklearn tests:
+            # svm/tests/test_sparse.py::test_error
+            # svm/tests/test_svm.py::test_bad_input
+            # for sklearn versions < 1.2 (i.e. without
+            # validate_params parameter checking)
+            # Without this, a segmentation fault with
+            # Windows fatal exception: access violation
+            # occurs
```
Comment (ethanglaser) [inline sklearnex/svm/svc.py:98]:
```suggestion
            # else if added to correct issues with
            # sklearn tests:
            # svm/tests/test_sparse.py::test_error
            # svm/tests/test_svm.py::test_bad_input
            # for sklearn versions < 1.2 (i.e. without
            # validate_params parameter checking)
            # Without this, a segmentation fault with
            # Windows fatal exception: access violation
            # occurs
            raise ValueError("C <= 0")
```

File: sklearnex/svm/nusvr.py [hunk 4]
```diff
@@ -65,6 +65,17 @@ def __init__(
...
+            raise ValueError("nu <= 0 or nu > 1")
+            # else if added to correct issues with
+            # sklearn tests:
+            # svm/tests/test_sparse.py::test_error
+            # svm/tests/test_svm.py::test_bad_input
+            # for sklearn versions < 1.2 (i.e. without
+            # validate_params parameter checking)
+            # Without this, a segmentation fault with
+            # Windows fatal exception: access violation
+            # occurs
```
Comment (ethanglaser) [inline sklearnex/svm/nusvr.py:78]:
```suggestion
            # else if added to correct issues with
            # sklearn tests:
            # svm/tests/test_sparse.py::test_error
            # svm/tests/test_svm.py::test_bad_input
            # for sklearn versions < 1.2 (i.e. without
            # validate_params parameter checking)
            # Without this, a segmentation fault with
            # Windows fatal exception: access violation
            # occurs
            raise ValueError("nu <= 0 or nu > 1")
```

File: sklearnex/svm/nusvr.py [hunk 5]
```diff
@@ -65,6 +65,17 @@ def __init__(
     def fit(self, X, y, sample_weight=None):
         if sklearn_check_version("1.2"):
             self._validate_params()
+        elif self.nu <= 0 or self.nu > 1:
```
Comment (samir-nasibli) [inline sklearnex/svm/nusvr.py:68]: I would prefer to add this check to the dispatch level, to the onedal_cpu/gpu_support functions, to faalback stock sklearn let him deal with error raising

File: sklearnex/svm/nusvr.py (hunk 4 above)
Comment (ethanglaser) [reply sklearnex/svm/nusvr.py:78]: Alternatively could put "Aligns with https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/svm/src/libsvm/svm.cpp#L3063" (or 3057 for C)

**Review by samir-nasibli — CHANGES_REQUESTED** [review]
Make sense add error raise on onedal4py level and just extend sklearnex level dispatching check for this conditions

**Comment by ethanglaser** [comment]
> Make sense add error raise on onedal4py level and just extend sklearnex level dispatching check for this conditions

Valid point but seeing as this is a fairly urgent fix for currently broken public CI would prefer not to mark "request changes" as it can slow down process. Can you provide a bit more specific detail on preferred option?

**Comment by samir-nasibli** [comment]
> > Make sense add error raise on onedal4py level and just extend sklearnex level dispatching check for this conditions
> 
> Valid point but seeing as this is a fairly urgent fix for currently broken public CI would prefer not to mark "request changes" as it can slow down process. Can you provide a bit more specific detail on preferred option?

If this is required for an urgent repair, then I remove the request change and you can work around like crazy. But in general, I would like to follow up to do a more in-depth study of the options and choose the best, since several different ones are offered

**Comment by samir-nasibli** [comment]
> > Make sense add error raise on onedal4py level and just extend sklearnex level dispatching check for this conditions
> 
> Valid point but seeing as this is a fairly urgent fix for currently broken public CI would prefer not to mark "request changes" as it can slow down process. Can you provide a bit more specific detail on preferred option?

If this is required for an urgent repair, then I remove the request change and we can merge it as lazy a work around. But in general, I would like to follow up to do a more in-depth study of the options and choose the best, since several different ones are offered

**Review by ethanglaser — APPROVED** [review]
Will want to not forget about taking a deeper look at underlying issues, but nice work on getting this fixed

**Comment by icfaust** [comment]
Private CI run: http://intel-ci.intel.com/ef3fc308-b385-f19d-a2cd-a4bf010d0e2e

**Comment by ethanglaser** [comment]
re-run: http://intel-ci.intel.com/ef3fd0e7-c417-f13f-b287-a4bf010d0e2e

**Comment by napetrov** [comment]
@mergify backport rls/2024.6.0-rls
